src/PhpImap/Mailbox.php

```diff
@@ -1340,7 +1340,7 @@ public function downloadAttachment(DataPartInfo $dataInfo, array $params, object
      *
      * @todo update implementation pending resolution of https://github.com/vimeo/psalm/issues/2619 & https://github.com/vimeo/psalm/issues/2620
      */
-    public function decodeMimeStr(string $string): string
+    public function decodeMimeStr(string $string, string $charsetPreferred = null): string
```

Comment:
this relates to a change made in 2231d5903ef448878b22b1b6cf2634cc57b2dc4c

Reply:
But I don't see that you are passing encoding here https://github.com/barbushin/php-imap/blob/develop/src/PhpImap/DataPartInfo.php#L108 

"imap_mime_header_decode" just returns "default" instead of correct one encoding...

Reply:
Previously you had this part 
https://github.com/barbushin/php-imap/commit/2231d5903ef448878b22b1b6cf2634cc57b2dc4c#diff-7f3ff3cf2fa5babe1eef4ffaa2ecc250L1388

And problem is with this encoding... "iso-8859-1"

Reply:
> But I don't see that you are passing encoding here https://github.com/barbushin/php-imap/blob/develop/src/PhpImap/DataPartInfo.php#L108
> 
> "imap_mime_header_decode" just returns "default" instead of correct one encoding...

this was changed in e40d88dadd8001f431cd7c578df67500af1d2805

Reply:
tl:dr; these changes (2231d59, e40d88d) were made and I don't know why 🤷‍♂️

Reply:
Well in short it fixed the issue I had and most likely the issue mentioned in #539 it's up to you to make proper fix. I tried to help :) But present code just does not work and that's a fact :)

Reply:
*is not the developer of this package, just contribute to it a whole bunch because it's used internally*